app-editors/scite/scite-4.1.2.ebuild

```diff
@@ -1,88 +1,110 @@
+SRC_URI="https://www.scintilla.org/${PN}${MY_PV}.tgz -> ${P}.tgz"
 
 LICENSE="HPND lua? ( MIT )"
 SLOT="0"
-KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd ~amd64-linux ~x86-linux"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux"
 IUSE="lua"
 
 RDEPEND="dev-libs/glib:=
 	x11-libs/cairo
-	x11-libs/gtk+:2
 	x11-libs/gdk-pixbuf
+	x11-libs/gtk+:3=
 	x11-libs/pango
 	lua? ( >=dev-lang/lua-5:= )"
+
 DEPEND="${RDEPEND}
 	virtual/pkgconfig"
 
 S="${WORKDIR}/${PN}/gtk"
 
+pkg_pretend() {
+	# older gcc is not supported
+	[[ "${MERGE_TYPE}" != "binary" && $(gcc-major-version) -lt 7 ]] && \
+		die "Sorry, but gcc earlier than 7.0 will not work for SCiTE."
+}
+
 src_prepare() {
 	sed -i "${WORKDIR}/scintilla/gtk/makefile" \
 		-e "s#^CXXFLAGS=#CXXFLAGS=${CXXFLAGS} #" \
 		-e "s#^\(CXXFLAGS=.*\)-Os#\1#" \
-		-e "s#^CC =\(.*\)#CC = $(tc-getCXX)#" \
 		-e "s#^CCOMP =\(.*\)#CCOMP = $(tc-getCC)#" \
 		-e "s#-Os##" \
 		|| die "error patching /scintilla/gtk/makefile"
 
-	sed -i "${WORKDIR}/scite/gtk/makefile" \
-		-e "s#-rdynamic#-rdynamic ${LDFLAGS}#" \
-		|| die "error patching /scite/gtk/makefile"
-
-	# repair and enhance the .desktop file
-	sed -i "${WORKDIR}/scite/gtk/SciTE.desktop" \
-		-e "s/^Encoding/#Encoding/" \
-		-e "s#text/plain#text/\*;application/xhtml+xml#" \
-		-e "s#^Categories=\(.*\)#Categories=Development;#" \
-		|| die "error patching /scite/gtk/SciTe.desktop"
-
 	sed -i "${S}/makefile" \
+		-e "s#-rdynamic#-rdynamic ${LDFLAGS}#" \
 		-e 's#usr/local#usr#g' \
 		-e 's#/gnome/apps/Applications#/applications#' \
 		-e "s#^CXXFLAGS=#CXXFLAGS=${CXXFLAGS} #" \
 		-e "s#^\(CXXFLAGS=.*\)-Os#\1#" \
-		-e "s#^CC =\(.*\)#CC = $(tc-getCXX)#" \
 		-e "s#^CCOMP =\(.*\)#CCOMP = $(tc-getCC)#" \
 		-e 's#${D}##' \
 		-e 's#-g root#-g 0#' \
 		-e "s#-Os##" \
 		|| die "error patching gtk/makefile"
 
+	# repair and enhance the .desktop file
+	sed -i "${S}/SciTE.desktop" \
+		-e "s#text/plain#text/\*;application/xhtml+xml#" \
+		-e "s#^Categories=\(.*\)#Categories=Development;#" \
+		|| die "error patching /scite/gtk/SciTe.desktop"
+
+	#  add the ebuild suffix as shell type for working with ebuilds
+	sed -i "${WORKDIR}/scite/src/perl.properties" \
+		-e "s#\*.sh;\*.bsh;#\*.ebuild;\*.sh;\*.bsh;#" \
+		|| die "error patching /scite/src/perl.prperties"
+
+	# it seems that pwd here is ${S}, but user patches are relative to ${workdir}
+	# Bug #576162
+
+	cd "${WORKDIR}" || die "couldn't cd into ${WORKDIR}"
 	eapply_user
 }
 
 src_compile() {
-	emake CC="$(tc-getCC)" LD="$(tc-getLD)" \
-			LDFLAGS="$(raw-ldflags)" AR="$(tc-getAR)" \
-			-C "${WORKDIR}/scintilla/gtk"
+	# prepare make options
+	local emake_pars="GTK3=1"
 
-	if use lua; then
-		emake
-	else
-		emake NO_LUA=1
+	if tc-is-clang; then
+		emake_pars+=" CLANG=1"
 	fi
+
+	if ! use lua; then
+		emake_pars+=" NO_LUA=1"
+	fi
+
+	emake CC="$(tc-getCC)" LD="$(tc-getLD)" \
+			LDFLAGS="$(raw-ldflags)" AR="$(tc-getAR)" \
+			-C "${WORKDIR}/scintilla/gtk" ${emake_pars}
```

Comment:
You're gonna need to add `" "` quotation marks around your `${emake_pars}` since the string can contain spaces.
`"${emake_pars}"`